bin/t1_and_mw_scan_descriptives.py

```diff
@@ -0,0 +1,157 @@
+    "Number of subs with posttraining/neuro directory: {}".format(
+        len(neuro_posttraining)
+    )
+)
+adie_con_split(neuro_posttraining)
+
+print("Number of T1s in ses-baseline: {}".format(len(t1_baseline)))
+adie_con_split(t1_baseline)
+print("Number of T1s in posttraining: {}".format(len(t1_posttraining)))
+adie_con_split(t1_posttraining)
+
+print("Number of MW funcs in ses-baseline: {}".format(len(mw_baseline)))
+adie_con_split(mw_baseline)
+print("Number of MW funcs in posttraining: {}".format(len(mw_posttraining)))
+adie_con_split(mw_posttraining)
+
+# Create dataframe
+# Row = sub
+# Colums = Baseline T1, Posttraining T1, Mindwandering baseline, Mind-wandering postrtaining
+
+# get sub names
+subs = [os.path.basename(i) for i in bids]
+df = pd.DataFrame(subs)
+df.rename(columns={0: "subjects"}, inplace=True)
+
+
+# Create dictionaries for columns
+# create dictionary where key = sub & val = number of niftis
+t1_bl = {}
+t1_pt = {}
+
+mw_bl = {}
+mw_pt = {}
+
+# Create empty columns
+df["t1_bl"] = 0
+df["t1_pt"] = 0
+df["mw_bl"] = 0
+df["mw_pt"] = 0
+
+# Create columns showing how many of each nifti each subject has
+for i in subs:
+    t1_bl[i] = [lst for lst in t1_baseline if i in lst]
+    df["t1_bl"].loc[df["subjects"] == i] = len(t1_bl[i])
+
+    t1_pt[i] = [lst for lst in t1_posttraining if i in lst]
+    df["t1_pt"].loc[df["subjects"] == i] = len(t1_pt[i])
+
+    mw_bl[i] = [lst for lst in mw_baseline if i in lst]
+    df["mw_bl"].loc[df["subjects"] == i] = len(mw_bl[i])
+
+    mw_pt[i] = [lst for lst in mw_posttraining if i in lst]
+    df["mw_pt"].loc[df["subjects"] == i] = len(mw_pt[i])
+
+
+# How many subjects are there with at least one T1 scan and both pre-and post-training mind wandering scans?
+full = df[
+    (df["mw_bl"] != 0)
+    & (df["mw_pt"] != 0)
+    & ((df["t1_bl"] != 0) | (df["t1_pt"] != 0))
+]
+print(
+    "Number of subjects with (i) both baseline and post-training MW and (ii) at least one T1: {}".format(
+        len(full)
+    )
+)
+
+
+# How many subjects with at least one T1 and a baseline MW scan?
+partial = df[(df["mw_bl"] != 0) & ((df["t1_bl"] != 0) | (df["t1_pt"] != 0))]
+print(
+    "Number of subjects with (i) baseline MW and (ii) at least one T1: {}".format(
+        len(partial)
+    )
+)
+
+print(df)
+
+# Save - change to 1 if you want to save out the file again
+save = 0
+# Merge with existing participants.tsv df
+
+if save == 1:
+    # import participants.tsv
+    participants = pd.read_csv(
+        "/research/cisc2/projects/critchley_adie/BIDS_data/participants.tsv",
+        sep="\t",
+    )
+    # change participant column to match
+    df.rename(columns={"subjects": "participant_id"}, inplace=True)
+    # remove 'sub-' from df
+    df["participant_id"] = df["participant_id"].str.replace("sub-", "")
+    # merge
+    df_master = df.merge(participants, on="participant_id", how="outer")
+    df_master.to_csv(
+        "/research/cisc2/projects/critchley_adie/BIDS_data/participants2.tsv"
```

Comment:
Avoid naming file like this. If this is for updating an existing file, why not over write the old file?
Output not fitting BIDS standard.